docs/tree_construction.md

```diff
@@ -7,6 +7,10 @@ For example, the rule `node: child1 child2` will create a tree node with two chi
 
 Using `item+` or `item*` will result in a list of items, equivalent to writing `item item item ..`.
 
+Using `item?` will return the item if it matched, or nothing.
+
+Using `[item]` will return the item if it matched, or the value `None`, if it didn't. It's possible to force `[]` to behave like `()?`, by using the `maybe_placeholders=False` option when initializing Lark.
+
```

Comment:
I am not sure I understand this right. Why I would use `maybe_placeholders=True` with `[something]`, instead of `(something)?` Can you give an usage example, i.,e., a grammar using `maybe_placeholders=False` versus `maybe_placeholders=True`?